### PR TITLE
feat:added productlisting page with functionality

### PR DESCRIPTION
**Pull request by Richa-rani02**
Functionality added:
1. added categorysort, pricesort(high and low), price slider,filter by rating ,searchfilter to productlisting page 
2. change navbar links based on routes
3. Linked category section of home page to productlisting page

### CODE REVIEW

**Review by sharathnair9999 — APPROVED** [review]

**Review by biswanath-tewari — APPROVED** [review]

**Review by Mohitkumar33 — APPROVED** [review]

File: src/pages/Home/Home.jsx [hunk 1]
```diff
@@ -7,7 +7,6 @@ import { useStateContext } from "../../context/stateContext";
 const Home = () => {
     const {state}=useStateContext();
     const {products}=state;
-    console.log(products);
```
Comment (deepak29-git) [inline src/pages/Home/Home.jsx:10]: remove console

File: src/pages/Products/Filter/Filter.css [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+  border-radius: 5px;
+}
+.clear-btn:hover{
+background-color: rgba(0, 0, 0, 0.12);
+}
+
+.input-group {
+    margin-bottom: var(--spacing-3);
+  }
+  label{
```
Comment (deepak29-git) [inline src/pages/Products/Filter/Filter.css:22]: use 'className' instead of selecting tag. don't use tag

File: src/utils/helper.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    let updatedList=[...productList];
+    if(searchProduct){
+        return updatedList.filter((product)=>product.title.toLowerCase().includes(searchProduct.toLowerCase()))
+    }
+    return updatedList;
+}
+export const sortByPrice=(productList,{sortByPrice})=>{
+    let updatedList=[...productList];
+        if (sortByPrice === "HIGH_TO_LOW") {
+            return updatedList.sort((a, b) => b["price"] - a["price"]);
```
Comment (deepak29-git) [inline src/utils/helper.js:12]: you can directly use price for example  b.price - a.price . you don't need to use [""]

File: src/utils/helper.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+          }
+          if (sortByPrice === "LOW_TO_HIGH") {
+            return updatedList.sort((a, b) => a["price"] - b["price"]);
```
Comment (deepak29-git) [inline src/utils/helper.js:15]: same here

**Review by deepak29-git — COMMENTED** [review]
do some changes which  I have suggested ,rest of looking good to me

**Pull request merged**